### PR TITLE
CBG-2736: Set replicator's initialStatus to allow coherent stats across rebalances

### PR DESCRIPTION
**Pull request by bbrks**
CBG-2736

A regression in #6113 prevented the `initialStatus` field of a replicator from being set. This prevented stats from persisting across replication rebalances and resetting to zero on a rebalance. Initialise the field with the status from a persisted checkpoint.

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `GSI=true,xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/1575/
  - Unrelated `TestInitializeIndexes`

### CODE REVIEW

File: db/active_replicator_common.go [hunk 1]
```diff
@@ -84,11 +84,18 @@ func newActiveReplicatorCommon(config *ActiveReplicatorConfig, direction ActiveR
 		checkpointID = PullCheckpointID(config.ID)
 	}
 
+	initialStatus, err := LoadReplicationStatus(config.ActiveDB.DatabaseContext, config.ID)
+	if err != nil {
+		// Not finding an initialStatus isn't fatal, but we should at least log that we'll reset stats when we do...
+		base.InfofCtx(context.TODO(), base.KeyReplicate, "Couldn't load initial replication status: %v - stats will reset", err)
```
Comment (torcolvin) [inline db/active_replicator_common.go]: use %w for err? and use apr.ctx?

Comment (bbrks) [reply db/active_replicator_common.go]: - I don't have `apr` or another `ctx` at this point (maybe I could pass it down from `NewActiveReplicator`?)
- `%w` doesn't work in logging, only in `fmt.Errorf`

Comment (bbrks) [reply db/active_replicator_common.go]: Passed through `ctx` from above, and included replication ID in log message.

File: db/active_replicator_common.go [hunk 2]
```diff
@@ -84,11 +84,18 @@ func newActiveReplicatorCommon(config *ActiveReplicatorConfig, direction ActiveR
 		checkpointID = PullCheckpointID(config.ID)
 	}
 
+	initialStatus, err := LoadReplicationStatus(config.ActiveDB.DatabaseContext, config.ID)
+	if err != nil {
+		// Not finding an initialStatus isn't fatal, but we should at least log that we'll reset stats when we do...
+		base.InfofCtx(ctx, base.KeyReplicate, "Couldn't load initial replication status for %q: %v - stats will be reset", config.ID, err)
```
Comment (torcolvin) [inline db/active_replicator_common.go]:
```suggestion
		base.InfofCtx(ctx, base.KeyReplicate, "Couldn't load initial replication status for %q: %w - stats will be reset", config.ID, err)
```

**Review by torcolvin — DISMISSED** [review]

**Review by torcolvin — APPROVED** [review]

**Pull request merged**